src/Verilator.cpp

```diff
@@ -420,13 +421,16 @@ static void process() {
         V3Dead::deadifyAll(v3Global.rootp());
     }
 
-    if (!v3Global.opt.lintOnly() && !v3Global.opt.xmlOnly() && v3Global.opt.oReloop()) {
-        // Reform loops to reduce code size
-        // Must be after all Sel/array index based optimizations
-        V3Reloop::reloopAll(v3Global.rootp());
-    }
-
     if (!v3Global.opt.lintOnly() && !v3Global.opt.xmlOnly()) {
+        // Merge conditionals
+        V3MergeCond::mergeAll(v3Global.rootp());
+
+        if (v3Global.opt.oReloop()) {
+            // Reform loops to reduce code size
+            // Must be after all Sel/array index based optimizations
+            V3Reloop::reloopAll(v3Global.rootp());
+        }
+
```

Comment:
If this becomes part of Reloop then original code is ok.
If this becomes its own stage then we should have a new optimization disable letter, and you should check that letter here.